### PR TITLE
Clean code for SVM

### PR DESCRIPTION
**Pull request by PetrovKP**
Clean code for thunder SVM on CPU and GPU

### CODE REVIEW

**Review by Alexsandruss — APPROVED** [review]

File: cpp/daal/src/algorithms/svm/svm_train_result.h [hunk 1]
```diff
@@ -135,7 +135,7 @@ class SaveResultTask
         {
             if (_alpha[i] != zero)
             {
-                DAAL_ASSERT(_cache->getDataRowIndex(i) <= services::internal::MaxVal<int>::get())
+                DAAL_ASSERT(_cache->getDataRowIndex(i) < _nVectors)
                 svIndices[iSV] = (int)_cache->getDataRowIndex(i);
```
Comment (agorshk) [inline cpp/daal/src/algorithms/svm/svm_train_result.h:139]: type of _cache->getDataRowIndex(i) is size_t?

File: cpp/oneapi/dal/algo/svm/backend/gpu/train_kernel_thunder_dpc.cpp [hunk 2]
```diff
@@ -76,7 +76,7 @@ static train_result call_daal_kernel(const context_gpu& ctx,
         desc.get_accuracy_threshold(),
         desc.get_tau(),
         desc.get_max_iteration_count(),
-        int64_t(desc.get_cache_size() * 1024 * 1024), // DAAL get in bytes
+        int64_t(desc.get_cache_size() * 1024.0 * 1024.0), // DAAL get in bytes
```
Comment (agorshk) [inline cpp/oneapi/dal/algo/svm/backend/gpu/train_kernel_thunder_dpc.cpp:79]: Does it possible overflow here?

File: cpp/daal/src/algorithms/svm/oneapi/svm_train_workset_oneapi.h [hunk 3]
```diff
@@ -116,6 +121,8 @@ struct TaskWorkingSet
             const size_t nCopy = utils::internal::min(nUpperSelect, nNeedSelect);
 
             context.copy(_wsIndices, _nSelected, _buffIndices, 0, nCopy, &status);
+            DAAL_CHECK_STATUS_VAR(status);
+
             _nSelected += nCopy;
```
[the remark this answers is not in the document]
Comment (PetrovKP) [reply cpp/daal/src/algorithms/svm/oneapi/svm_train_workset_oneapi.h:126]: `_nSelected` < `nVectors` always => `_nSelected` < `MAX(size_t)`

**Review by agorshk — APPROVED** [review]

**Pull request merged**